lib/langchain/tool/file_system/file_system.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Langchain::Tool
+  class FileSystem < Base
+    #
+    # A tool that wraps the Ruby file system classes.
+    #
+    # Usage:
+    #    file_system = Langchain::Tool::FileSystem.new
+    #
+    NAME = "file_system"
+    ANNOTATIONS_PATH = Langchain.root.join("./langchain/tool/#{NAME}/#{NAME}.json").to_path
+
+    description <<~DESC
+      A tool that wraps the Ruby File class, for interacting with the file system.
+    DESC
+
+    def list_directory(directory_path:)
+      Dir.entries(directory_path)
+    rescue Errno::ENOENT
+      "No such directory: #{directory_path}"
```

Comment:
Do we want this message just returned, or logged, or both?

Reply:
Definitely returned -- it communicates to the LLM what the issue is.

I've been thinking about a clean logging system when the tools are executed, something like:
![image](https://github.com/andreibondarev/langchainrb/assets/541665/3b7ea4f7-dd43-4f63-999a-92d2ef1f9c57)